core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -501,15 +502,35 @@ public static String cleanPath(String path) {
     }
 
     public static Set<String> splitStringByCommaToSet(final String s) {
-        return splitStringToSet(s, ',');
+        return splitStringToSet(s, ',', false);
     }
 
     public static String[] splitStringByCommaToArray(final String s) {
         if (s == null || s.isEmpty()) return Strings.EMPTY_ARRAY;
         else return s.split(",");
     }
 
-    public static Set<String> splitStringToSet(final String s, final char c) {
+    public static Set<String> splitStringByCommaAndTrim(final String s) {
```

Comment:
actually, i see 9 uses of the original method, and none look like they would want to retain whitespace. i think we should just change the behavior?